detection.py

```diff
@@ -0,0 +1,56 @@
+import cv2
+
+MIN_AREA = 1000  # площадь меньше, которой обнаруженные маски будут считаться дефектными
+
+
+def get_edges(image):
+    gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    blur = cv2.GaussianBlur(gray, (1, 1), 0)
+    edged = cv2.Canny(blur, 100, 400)
+    kernel = cv2.getStructuringElement(cv2.MORPH_RECT, (7, 7))
+    return cv2.morphologyEx(edged, cv2.MORPH_CLOSE, kernel)
+
+
+def get_cut_images(image, closed, contours):
+    height, width = image.shape[:2]
+    for cnt in contours:
+        if cv2.contourArea(cnt) > (height - 200) * (width - 200):
+            x, y, w, h = cv2.boundingRect(cnt)
+            image = image[y + 50:y + h - 50, x + 50:x + w - 50]
+            closed = closed[y + 50:y + h - 50, x + 50:x + w - 50]
+            contours, _ = cv2.findContours(closed.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
```

Comment:
```suggestion
            contours, _ = cv2.findContours(closed, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
```
> [Since opencv 3.2 source image is not modified by this function.](https://docs.opencv.org/4.x/d3/dc0/group__imgproc__shape.html#gadf1ad6a0b82947fa1fe3c3d497f260e0)

Reply:
Исправила